src/actions/proxy.ts

```diff
@@ -0,0 +1,129 @@
+import { Constructor, Tinlake } from '../types';
+import { waitAndReturnEvents, executeAndRetry } from '../ethereum';
+const abiCoder = require('web3-eth-abi');
+import BN from 'bn.js';
+
+function ProxyActions<ActionsBase extends Constructor<Tinlake>>(Base: ActionsBase) {
+
+  return class extends Base implements IProxyActions {
+
+    shelf = this.contracts['SHELF'].address;
```

Comment:
i don't think you need declare these as class variables, bc contracts is already a class variable. Would store the same information twice